### DISCUSSION TITLE
Python 3.9 Post Install fails

### DISCUSSION THREAD

**Discussion by rwachs** [Homebrew/discussions#746] [Getting started]
I am attempting to get a working version of Python 3.X installed on my iMac currently running 10.15.7 (Catalina) to run one script. I have brew installed as well as Python 3.9. When I try to take the next step and run the post install I get an error," 

Warning: The post-install step did not complete successfully
You can try again using:
  brew postinstall python@3.9

Adding  the "@3.9" still doesn't show a completed process. The Debug script generates the following:

brew postinstall --debug node
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/node.rb
==> Postinstalling node
Warning: The post-install step did not complete successfully
You can try again using:
  brew postinstall node
==> An exception occurred within a child process:
  Errno::ENOENT: No such file or directory @ rb_file_s_stat - /usr/local/Cellar/node/15.8.0/libexec/lib/node_modules/npm

/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:1305:in `stat'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:1305:in `lstat'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:1350:in `copy'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:478:in `block in copy_entry'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:1484:in `wrap_traverse'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:475:in `copy_entry'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:453:in `block in cp_r'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:1557:in `block in fu_each_src_dest'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:1571:in `fu_each_src_dest0'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:1555:in `fu_each_src_dest'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/fileutils.rb:452:in `cp_r'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/node.rb:70:in `post_install'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1058:in `block (2 levels) in run_post_install'
/usr/local/Homebrew/Library/Homebrew/formula.rb:900:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1057:in `block in run_post_install'
/usr/local/Homebrew/Library/Homebrew/utils.rb:532:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1046:in `run_post_install'
/usr/local/Homebrew/Library/Homebrew/postinstall.rb:22:in `<main>'

Any advice on how to proceed and fix this issue is most appreciated!!!!

**Comment by gromgit** [comment]
> ```
> Warning: The post-install step did not complete successfully
> You can try again using:
> brew postinstall python@3.9
> ...
> brew postinstall --debug node
> ```

Are you trying to post-install python or node?

For node, try `brew reinstall node`, as it looks like your npm directory got nuked somehow.

For python, you'd have to show us the correct debug output.

**Comment by rwachs** [comment]
Well, thank you. I tried the reinstall node command and I think I am at
least one step closer. I am not conversant in python (obviously) and am
just trying to get an install to run one script each month. Quite an
adventure! Thank you again for your time and assistance!
